examples/api/lib/widgets/pop_scope/pop_scope.0.dart

```diff
@@ -62,8 +62,12 @@ class _PageTwo extends StatefulWidget {
 }
 
 class _PageTwoState extends State<_PageTwo> {
-  void _showBackDialog() {
-    showDialog<void>(
+  /// Shows a dialog and resolves to true when the user has indicated that they
+  /// want to pop.
+  ///
+  /// A null indicates a desire not to pop.
```

Comment:
Where's null introduced?

Reply:
showDialog can return a Future that resolves to null. Dismissing the dialog (such as by tapping the barrier) will resolve to null. In those cases, the user doesn't want to leave the page.

I'll change that to "A return value of null indicates a desire not to pop, such as when the user has dismissed the modal without tapping a button."